### PR TITLE
[luci-interpreter]Negative Tests for AveragePool2D kernel

### PR DESCRIPTION
**Pull request by struss**
For #1623

This commit add negative tests on `AveragePool2D` kernel on luci-interpreter.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

**Comment by jinevening** [comment]
I'm not sure it is ok to check the things (type/shape/quantparam) inside the kernels, because this will make interpreter slow.

IMHO, those checks should be performed in the `import` stage. 

@s-barannikov What do you think of this?

**Comment by seanshpark** [comment]
> IMHO, those checks should be performed in the import stage.

@jinevening , @s-barannikov , there is already merged #3756 . Please check this too.

**Comment by s-barannikov** [comment]
@jinevening 
I don't have strong opinion on that.

On one hand, things like data type should be correct at the time the interpreter is called. `assert` should be enough for such checks. On the other, shapes of outputs are evaluated as operators are executed, and cannot always be statically known to be guaranteed to be correct.

I don't think that adding any checks of the form `if (!condition) throw exception` would slow down the interpreter -- such checks only add are a couple of processor instructions to the hot path, until the `condition` involves some significant amount of computations (comparable to the complexity of the operation itself). But at the same time, adding too many checks would make the code less understandable, we certainly should not check _everything_: TFLite has its own code well tested, and we should not test what is already tested.

In short, I don't mind adding such checks. But it may be worth to introduce some macro analogous to TFLIte's `TFLITE*CHECK*` to reduce the number of `if` statements to make the code more readable.

**Comment by s-barannikov** [comment]
@seanshpark Is the requirement for the negative tests only a [necessary] formality?

**Comment by seanshpark** [comment]
>  Is the requirement for the negative tests only a [necessary] formality?

Um... not sure about `[necessary] formality`  :)
What I know is that our project process requires the number of (new) negative test cases should be higher than (new) positive test cases.

**Comment by jinevening** [comment]
I'm ok with this change as @s-barannikov approved. Please go ahead.

**Comment by s-barannikov** [comment]
I mean, if it is a formality, we could add some dummy tests. For example, try to execute kernels with unsupported data type, such as FLOAT64.

**Comment by seanshpark** [comment]
> if it is a formality, we could add some dummy tests. For example, try to execute kernels with unsupported data type, such as FLOAT64.

Yes, obvious tests like your example is also OK I guess :)
Personally I think they may alarm un-intended changes in the future.
